ryan_savill/03_background_subtraction/readme.md

```diff
@@ -0,0 +1,281 @@
+
+I for instance have chosen to work with data in the form of dataframes (if you use [R](https://www.r-project.org/) or [matlab](https://de.mathworks.com/products/matlab.html) this might be familiar to you) with a python library called [pandas](https://pandas.pydata.org/). It comes pre-installed with anaconda and makes my life a little easier as all columns have names unlike numpy arrays, which could create a lot of confusion. When I want to perform some kind of analysis on my dataframes often functions will output results as numpy arrays, so I have to turn them back into dataframes and sometimes the dataframes need to be converted into arrays as input for some libraries. Functions help me everyday, because the small lines of code that convert datatypes or reshape lists for specific functions really sum up if you have to write them everytime. Also these lines of code make the files ineligable, whereas functions are just neat one-liners which I can easily read! But how can you make a library? I'll show you a simple way now and a more in depth way later:
+
+### How to Build Your Own Simple Library
+- First you have to make a python file and name it something recogniseable (like: img_analysis_functs). 
+- Then you have to add the functions you want to include in your library. Make sure that each function also contains all the import statements of the libraries it uses, otherwise you might be met with errors later on! 
+- And believe it or not that was it! 
+
+The important thing is that this python file is in the same folder as the python file you want to use your library in. If this is the case you should be able to import it like this:
+
+
+```python
+import img_analysis_functs as iaf
+
+io.imshow(iaf.subtract_background(tribolium))
+```
+
+
+    
+![png](output_17_2.png)
+    
+
+
+## Difference of Gaussian
+This is another method of highlighting nuclei, even if it is not described as background subtraction but rather [feature enhancement](https://en.wikipedia.org/wiki/Difference_of_Gaussians). In essence what is done is that we generate two gaussian blurred images with differing sizes of gaussians. The features we want to highlight should be in between the two sizes of the gaussians. Afterwards we subtract the larger blurred image from the smallller one. In a way this combines the noise removal of the small gaussian blurred image with a background subtraction represented by the larger gaussian blurred image. If filters are a common concept for you you can think of it as a bandpass filter that highlights thefeatures in between the two sizes. Since gaussian blur is a simple function included in scikit image let's create this filter ourselves:
```

Comment:
```suggestion
This is another method of highlighting nuclei, even if it is not described as background subtraction but rather [feature enhancement](https://en.wikipedia.org/wiki/Difference_of_Gaussians). In essence, it generates two Gaussian blurred images with different sigma for the Gaussian blur filter. The features we want to highlight should hafe a size between the two sigma values of the Gaussian blurs. Afterwards, we subtract the larger blurred image from the smallller one. In a way this combines the noise removal of the small Gaussian blurred image with a background subtraction represented by the larger Gaussian blurred image. If filters are a common concept for you, you can think of it as a bandpass filter that highlights the features in between the two sizes. Since [Gaussian blur](https://scikit-image.org/docs/dev/api/skimage.filters.html#skimage.filters.gaussian) is a function included in scikit image let's create this filter ourselves:
```

Capital Gauss because it was a person.